GalaxyZooTouchTable/GalaxyZooTouchTable/ViewModels/ClassificationPanelViewModel.cs

```diff
@@ -103,22 +121,37 @@ public ClassificationPanelViewModel(Workflow workflow, UserConsole console)
 
         private void OnPropertyRaised(string propertyname)
         {
-            PropertyChanged(this, new PropertyChangedEventArgs(propertyname));
+            if (PropertyChanged != null)
+                PropertyChanged(this, new PropertyChangedEventArgs(propertyname));
         }
 
         private void LoadCommands()
         {
             SelectAnswer = new CustomCommand(ChooseAnswer, CanSelectAnswer);
             SubmitClassification = new CustomCommand(SendClassification, CanSendClassification);
             ShowCloseConfirmation = new CustomCommand(ToggleCloseConfirmation);
-            EndSession = new CustomCommand(CloseClassifier);
+            CloseClassifier = new CustomCommand(OnCloseClassifier);
+            OpenClassifier = new CustomCommand(OnOpenClassifier);
+        }
+
+        private void OnOpenClassifier(object sender)
+        {
+            ClassifierOpen = true;
+            ActiveUsers.Add(User);
         }
 
-        private async void CloseClassifier(object sender)
+        private void OnCloseClassifier(object sender)
         {
-            await Console.Classifier.MoveClassifier();
-            Console.MoveButton();
-            Console.ClassifierOpen = !Console.ClassifierOpen;
+            ClassifierOpen = false;
+            CloseConfirmationVisible = false;
+            foreach (TableUser user in ActiveUsers)
```

Comment:
I can for sure change the name to `ActiveUser`, but I will get a collection modified error if trying that slick method because the collection is modified mid-enumeration.

Reply:
There are some interesting ways to do array filtering I found here: https://stackoverflow.com/questions/1912128/filter-an-array-in-c-sharp